### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by DanielLenz**
Due to the current issues with OS X builds, I've been doing all the development/testing in a docker container. This PR adds a Dockerfile to the healpy repo, some instructions are currently at the beginning of the Dockerfile.

If people are interested, we can also document this elsewhere.

I've also pushed the file to https://hub.docker.com/r/healpy/python3.6-healpy/, please ping me with your dockerhub username to be added to the organization.

Addresses https://github.com/healpy/healpy/issues/494

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    gcc \
+    g++ \
+    git \
+    autoconf \
+    automake \
+    libtool \
+    pkg-config
+
+# Copy the local healpy copy into the docker container
+COPY . /home/healpy
```
Comment (zonca) [inline Dockerfile]: I think it would be useful here to clone master from Github and build that.

**Comment by zonca** [comment]
thanks @DanielLenz , is https://hub.docker.com/r/healpy/python3.6-healpy/ automatically building when we tag a release or make a commit on master?

**Comment by DanielLenz** [comment]
Thanks @zonca , these are great suggestions!

> thanks @DanielLenz , is https://hub.docker.com/r/healpy/python3.6-healpy/ automatically building when we tag a release or make a commit on master?

Not yet! I won't find the time this weekend, but look forward to implementing that as soon as possible.

**Comment by DanielLenz** [comment]
I've just rebased this against `master`, and the [docker repo](https://hub.docker.com/r/healpy/healpy/) is set to auto build, even though I haven't tested that yet.

Could you double check this once the `Dockerfile` is in healpy/master, to see if it works in the live environment?

**Comment by DanielLenz** [comment]
Should we proceed with this, @zonca ? Let me know if you have any more comments or suggestions.

**Comment by DanielLenz** [comment]
I've updated this to python3.7 (had some issues with the installation on the Docker Python 3.6 image), and also changed it to copy the local healpy folder into the docker container. I think the dominant use case for this is development, and the workflow

```bash
> git checkout feature
> docker build ...
> docker run (mounting the local copy)
> <in container> edit, test, install, etc
> exit container, retain all the changes made in it
> git commit/push/merge etc
```

**Comment by zonca** [comment]
I think this is very useful for development
I wonder if we should have instead another version of this for docker hub instead bundles the latest healpy version, so that people which have some environment issues could use a docker container instead of installing healpy directly, we could have a container tagged with the healpy version.
Anyway I will merge this for now.

**Pull request merged**